# interface-param.yaml

# Created on: Apr 12, 2025

# Author: Gabriel Toffanetto Fran√ßa da Rocha

# Laboratory of Autonomous Mobility (LMA)
# School of Mechanical Engineering (FEM)
# University of Campinas (Unicamp)

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
  
 vilma_interface:
    ros__parameters:
        ma_timer_period_ms: 10
        ma_sleep_period_min: 2

        pc_udp_port: 5001
        ma_udp_port: 5001
        udp_timeout_ms: 4
        ma_ip: "192.168.140.3"

        autonomous_shift_enable: false

        kp_vel: 0.0015
        ki_vel: 0.0
        kd_vel: 0.0
        brake_deadband: -0.1
        speed_reference_ramp_rate: 3.0
        brake_user_pressure_set_emergency: 10.0

        max_steering_tire_angle_rad: 25.27 * 3.14156 / 180.0
        max_gas_value: 1.0
        max_brake_value: 1.0
        max_speed_km_h: 60.0
        
        joystick_command_time_validity_ms: 500
        communication_timeout_ms: 1000
        autoware_command_time_validity_ms: 250